drv/BigArrayBigList.drv

```diff
@@ -737,6 +743,75 @@ public class BIG_ARRAY_BIG_LIST KEY_GENERIC extends ABSTRACT_BIG_LIST KEY_GENERI
 			return new SubListSpliterator();
 		}
 
+		boolean contentsEquals(KEY_GENERIC_TYPE[][] otherA, long otherAFrom, long otherATo) {
+			if (a == otherA && from == otherAFrom && to == otherATo) return true;
+			if (otherATo - otherAFrom != size64()) {
+				return false;
+			}
+			long pos = from, otherPos = otherAFrom;
+			// We have already assured that the two ranges are the same size, so we only need to check one bound.
+			// TODO When minimum version of Java becomes Java 9, use the Arrays.equals which takes bounds, which is vectorized.
+			// Make sure to split out the reference equality case when you do this.
+#if KEY_CLASS_Object
+			while(pos < to) if (! java.util.Objects.equals(BigArrays.get(a, pos++), BigArrays.get(otherA, otherPos++))) return false;
```

Comment:
These need to be made backwards loops because they are faster then forward loops of `java.util.Arrays` apparently

Reply:
Done